pkg/virt-launcher/virtwrap/network/ndp/ndp.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package ndp
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/mdlayher/ndp"
+
+	"kubevirt.io/client-go/log"
+)
+
+const (
+	routerAdvertisementMaxLifetime = 65535 * time.Second // check RFC 4861, section 4.2; 16 bit integer.
+)
+
+type RouterAdvertisementDaemon struct {
+	raOptions []ndp.Option
+	ndpConn   *NDPConnection
+}
+
+func RouterAdvertisementDaemonFromFD(openedFD *os.File, ifaceName string, ipv6CIDR string) (*RouterAdvertisementDaemon, error) {
```

Comment:
I would change to CreateRouterAdvertisementServerFromFD

Reply:
https://github.com/kubevirt/kubevirt/pull/4581/commits/014fadcf3e4d807d9a21db6cbfa3dc3f96bdaa07#diff-7cea0cc72a89ff86def413e0982f49cbbb527157a092cd507a336ced23b7ce2aR516